### PR TITLE
fix: use `implements` for interface extension.

### LINKED ISSUE #263: Interfaces that extend other interfacesin TS should implement them in dart
Currently the following TypeScript code

```TypeScript
interface A {...}

interface B extends A {...}
```
Gets transpiled to the following Dart
```dart
abstract class A {...}
abstract class B extends A {...}
```
But we should really be using implements instead of extends here to more accurately emulate what's going in TypeScript.
For example, with extends in Dart it is not possible for a class to extend multiple parent classes but in TypeScript an interface can implement multiple parent interfaces. Currently,
```TypeScript
interface A {...}

interface B {...}

interface A extends A, B {...}
```
becomes
```dart
abstract class A {...}
abstract class B {...}

abstract class C extends A, B {...}
```
Which is not valid Dart, but

```dart
abstract class A {...}
abstract class B {...}

abstract class C implements A, B {...}
```
Would be correct.

### PR DESCRIPTION
**Pull request by mprobst**
TypeScript interfaces are represented as abstract classes in Dart. Abstract
classes can either implement or extend super classes, but an interface extending
multiple super interfaces should be represented by `implements`, as Dart does
not have multiple inheritance for abstract classes.

Fixes #263.

### CODE REVIEW

File: test/declaration_test.ts [hunk 1]
```diff
@@ -102,23 +102,27 @@ describe('classes', () => {
       expectTranslate('class X { constructor() { } }').to.equal(' class X { X ( ) { } }');
     });
     it('supports parameter properties', () => {
+      expectTranslate('class X { c: number; \n' +
```
Comment (mprobst) [inline test/declaration_test.ts]: FYI these changes are unrelated, I just wanted to fix the lines >> 100 chars here.

File: test/declaration_test.ts [hunk 2]
```diff
@@ -102,23 +102,27 @@ describe('classes', () => {
...
-  it('should translate interfaces',
+  it('translates interfaces to abstract classes',
      () => { expectTranslate('interface X {}').to.equal(' abstract class X { }'); });
-  it('should support extends', () => {
-    expectTranslate('interface X extends Y, Z {}').to.equal(' abstract class X extends Y , Z { }');
+  it('translates interface extends to class implements', () => {
+    expectTranslate('interface X extends Y, Z {}')
+        .to.equal(' abstract class X implements Y , Z { }');
   });
   it('should support implements', () => {
```
Comment (vicb) [inline test/declaration_test.ts]: _Not directly related to this PR:_
Seems like the 2 following tests are duplicates (from the 'classes' group)

**Comment by vicb** [comment]
LGTM with one comment about duplicate tests (not directly related to this PR)

File: test/declaration_test.ts (hunk 2 above)
Comment (mprobst) [inline test/declaration_test.ts]: Good catch, done.
